src/main.rs

```diff
@@ -355,3 +337,26 @@ fn build_config(version: &str) -> Config {
 
     config
 }
+
+fn init_config(config: &Config) -> CommandMode {
+    config.auth.as_ref().map(|auth_cfg| {
+        let ref secret   = auth_cfg.client_secret;
+        let ref auth_p12 = auth_cfg.p12_path;
+        let ref dev_p12  = config.device.p12_path;
+
+        if secret.is_some() == auth_p12.is_some() {
```

Comment:
Error message should specify that this is a XOR and not an OR because human language is ambiguous.